### ISSUE TITLE
Model.objects.get_by_attribute() returning empty querysets

### ISSUE THREAD

**Issue by strikaco** [evennia/evennia#1627]
#### Brief summary of issue / Description of requested feature:
I'm trying to look up an object by attribute. As I understand it, every class' object manager has a get_by_attribute() method-- but it doesn't seem to work (at least for the Character and ObjectDB models).

#### Steps to reproduce the issue / Reasons for adding feature:
    # Version info
    >>> import evennia
    >>> evennia.__version__
    '0.7.0 (rev 5335f217)'

    # Just to prove that the attributes *do* exist!
    >>> from typeclasses.characters import Character
    >>> [(x,y) for (x,y) in Character.objects.all().values_list('db_attributes__db_key', 'db_attributes__db_value') if y == 'johnny']
    [(u'slug', u'johnny')]

    # Let's try it the documented way
    >>> Character.objects.get_by_attribute(key='slug', value='johnny')
    <QuerySet []>

    # Let's try bypassing the manager and referencing the attributes directly
    >>> Character.objects.filter(db_attributes__db_key='slug', db_attributes__db_value='johnny') 
    <QuerySet []>

    # Is it an encoding issue?
    >>> Character.objects.filter(db_attributes__db_key=u'slug', db_attributes__db_value=u'johnny')
    <QuerySet []>

    # Now I'm just getting desperate
    >>> Character.objects.filter(db_attributes__db_key__iexact=u'slug', db_attributes__db_value__iexact=u'johnny')
    <QuerySet []>

    # Here's the query being executed, still poking around at this
    >>>str(Character.objects.filter(db_attributes__db_key='slug', db_attributes__db_value='johnny').query)
    'SELECT "objects_objectdb"."id", "objects_objectdb"."db_key", "objects_objectdb"."db_typeclass_path", "objects_objectdb"."db_date_created", "objects_objectdb"."db_lock_storage", "objects_objectdb"."db_account_id", "objects_objectdb"."db_sessid", "objects_objectdb"."db_location_id", "objects_objectdb"."db_home_id", "objects_objectdb"."db_destination_id", "objects_objectdb"."db_cmdset_storage" FROM "objects_objectdb" INNER JOIN "objects_objectdb_db_attributes" ON ("objects_objectdb"."id" = "objects_objectdb_db_attributes"."objectdb_id") INNER JOIN "typeclasses_attribute" ON ("objects_objectdb_db_attributes"."attribute_id" = "typeclasses_attribute"."id") WHERE ("objects_objectdb"."db_typeclass_path" = typeclasses.characters.Character AND "typeclasses_attribute"."db_value" = gAJVBmpvaG5ueXEBLg== AND "typeclasses_attribute"."db_key" = slug)'

#### Error output / Expected result of feature
Surely this is not intended behavior? I got the impression it should return a queryset containing the objects with this k:v pair in its attributes.

#### Extra information, such as Evennia revision/repo/branch, operating system and ideas for how to solve / implement:
For what it's worth (and if it impacts anything) I've been doing a lot of getting/setting of most of these attributes via Django views, not in-game. I'm wondering if there is some sort of cache that isn't flushing, though the attributes set via the web interface do show up in the shell, so I'd find it unlikely that this would impact it.

Thanks for any assistance!

**Comment by TehomCD** [comment]
Seems to work in evennia shell for me:
```
(arx) c:\arx\arx>evennia shell
Python 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:53:40) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from typeclasses.characters import Character
>>> Character.objects.get_by_attribute(key="fealty", value="Velenosa")
<QuerySet [<Character: Anarae>, <Character: Jaenelle>, <Character: Deva>, <Character: Ferrando>, <Character: Leo>, <Character: Audric>, <Character: Esera>, <Character: Calista>, <Character: Isolde>, <Character: Talen>, <Character: Niccolo>, <Character: Alessandra>, <Character: Malrico>, <Character: Belladonna>, <Character: Viviana>, <Character: Caprice>, <Character: Valencia>, <Character: Ophelia>, <Character: Antonio>, <Character: Hadrian>, '...(remaining elements truncated)
...']>
>>> import evennia
>>> evennia.__version__
'0.7.0 (rev 9e7969f)'
>>>
```

**Comment by Griatch** [comment]
@strikaco `get_by_attribute` is a Django manager method so it should not be affected by handler-caches. 

It works for me too. In-game I did

    @py self.db.foo = bar

Then, in `mygame`: 

    $ evennia shell
    > from typeclasses.characters import Character 
    > Character.objects.get_by_attribute(key='foo', value='bar')    
    <QuerySet [<Character: Griatch>]>

If you try this exact sequence, does it work for you then?

**Comment by strikaco** [comment]
Following your examples, it does work.

Oddly enough I tried the same exercise I posted with a different key:value pair and it also worked.

What still doesn't work is if I search for the specific key "slug" as in my example. I don't understand why (and didn't see any references to slugs being used for anything else in code, so it shouldn't be any sort of conflict). I'm running stock EV with no crazy contribs or extensions.

The only thing about this attribute that might be peculiar is that it was set within a transaction in a View. Would this have interfered with anything? I know attributes are more than simple ModelFields.

(If not, we can close this issue. I have since worked around it by creating an identical alternative field and searching on it instead.)

**Comment by strikaco** [comment]
Ok, I've figured this one out.

Somehow I had set some properties on objects that had not persisted to DB (still suspect an aborted transaction might have interfered somehow) via a web-based character creation screen. Properties were supposed to have been set during at_object_creation, but it seems if you never actually enter the game using that character, the function never gets called.

Moving all initial property setters to AOC and calling it manually fixes the problem I was having and this function works as expected.
